tests/modules/configs/test_config.py

```diff
@@ -0,0 +1,86 @@
+import os
+import pytest
+import sys
+
+from jsonschema import ValidationError
+from logging import Logger
+
+sys.path.append(os.getcwd())
+
+from src.modules.configs.config import ApplicationConf, ProjectConf, OperatorConf, \
+    MergedConf, validate_dict
+from src.modules.configs.schemas import application, project, operator
+from src.modules.global_vars import APPLICATION_DIR, CONFIG_RULES
+from tests.data.configs import validate_conf, err_msg, config_rule
+
+current_path = os.path.join(os.path.dirname(__file__))
+test_data_path = os.path.join(os.path.dirname(os.path.dirname(current_path)),
+                              "data", "configs")
+
+
+@pytest.mark.parametrize("conf, schema_file, err_msg", [
+    [validate_conf.application.get("unrecognized_key"), application,
+     err_msg.validate_conf.get("unrecognized_key")],
+    [validate_conf.project.get("unrecognized_key"), project,
+     err_msg.validate_conf.get("unrecognized_key")],
+    [validate_conf.operator.get("unrecognized_key"), operator,
+     err_msg.validate_conf.get("unrecognized_key")],
+
+    [validate_conf.application.get("mandatory_key1"), application,
+     err_msg.validate_conf.get("mandatory_key")],
+    [validate_conf.application.get("mandatory_key2"), application,
+     err_msg.validate_conf.get("mandatory_key")],
+
+    [validate_conf.application.get("invalid_data_type"), application,
+     err_msg.validate_conf.get("invalid_data_type")],
+    [validate_conf.project.get("invalid_data_type"), project,
+     err_msg.validate_conf.get("invalid_data_type")],
+    [validate_conf.operator.get("invalid_data_type"), operator,
+     err_msg.validate_conf.get("invalid_data_type")],
+
+    [validate_conf.application.get("too_long_value"), application,
+     err_msg.validate_conf.get("too_long_value")],
+    [validate_conf.project.get("too_long_value"), project,
+     err_msg.validate_conf.get("too_long_value")],
+    [validate_conf.operator.get("too_long_value"), operator,
+     err_msg.validate_conf.get("too_long_value")],
+
+    [validate_conf.application.get("non_mandatory_key"), application, ""],
+    [validate_conf.project.get("non_mandatory_key"), project, ""],
+    [validate_conf.operator.get("non_mandatory_key"), operator, ""],
+
+    [validate_conf.application.get("successful_validation"), application, ""],
+    [validate_conf.project.get("successful_validation"), project, ""],
+    [validate_conf.operator.get("successful_validation"), operator, ""],
+])
+def test_validate_dict(conf, schema_file, err_msg):
+    try:
+        validate_dict(conf, schema_file)
+    except ValidationError as ex:
+        assert err_msg in str(ex)
+    else:
+        assert validate_dict(conf, schema_file) is None
+
+
+class TestConf:
+    """Class to test functions of Conf class"""
+
+    @pytest.mark.parametrize("conf_path, conf_file, real_conf_path", [
+        [os.path.join(test_data_path, "application.json"), "application.json",
```

Comment:
You can move this list to class variable e.g. `conf_list` and use it in fixture as `TestConf.conf_list`. If data gets too bulky you can create locally `TestConfData` and add only variables there.

Reply:
tried to split to separate test functions